assembly/Int/Int32.ts

```diff
@@ -23,9 +23,9 @@ export class Int32 extends AbstractInt<i32> {
     }
 
     /** Instantiates new Int32 from u8[] SCALE encoded bytes */
-    static fromU8a (value: u8[]): Int32 {
-        assert(value.length <= BIT_LENGTH.INT_32, 'Int32: Invalid bytes provided');
-        var res = Bytes.toUint<u32>(value, BIT_LENGTH.INT_32);
+    static fromU8a (value: u8[], index: i32 = 0): Int32 {
+        assert(value.length - index > 0, 'Int32: Empty bytes array provided');
```

Comment:
To be absolutely correct, shouldn't this assert be:
```JavaScript
assert(value.length - index > 0 && value.length - index <= BIT_LENGTH.INT_32`, 'Int32: Invalid bytes provided');
```

Same for the other types. We should check not only that the index is bigger than the `value.length` but we should check that it is low enough so that the SCALE type can be read, maybe?

Reply:
resolved it